api/controller/spotify_controller.js

```diff
@@ -122,19 +122,44 @@ selectWeightedRandoms = (data_struct, max_size) => {
   return selectedRandoms;
 }
 
+mergeSeedObjs = (obj1, obj2) => {
+  let merged = {};
+
+  for (var key in obj1) {
+    if (obj2[key] == null) { // keys in obj1 not in obj2
+      merged[key] = obj1[key];
+    } else { // keys in obj1 and obj2
+      let merged_weight = obj1[key].weight + obj2[key].weight;
+      let merged_value = { weight: merged_weight, last_accessed: 0 };
+      merged[key] = merged_value;
+    }
+  }
+
+  for (var key in obj2) { // keys in obj2 not in obj1
```

Comment:
`return Object.assign(obj2, merged);`

Reply:
true, gonna do `return {...obj2, ...merged};` instead of no objections